service/arm_templates/master_nodes_setup.json

```diff
@@ -0,0 +1,169 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "networkInterfacesAPIVersion": {
+      "type": "string",
+      "defaultValue": "2016-09-01",
+      "metadata": {
+        "description": "API version used by the Microsoft.Network/networkInterfaces resource."
+      }
+    },
+    "virtualMachinesAPIVersion": {
+      "type": "string",
+      "defaultValue": "2016-08-30",
+      "metadata": {
+        "description": "API version used by the Microsoft.Compute/virtualMachines resource."
+      }
+    },
+    "clusterID": {
+      "type": "string"
+    },
+    "storageAccountName": {
+      "type": "string"
+    },
+    "masterCount": {
+      "type": "int"
+    },
+    "masterSubnetID": {
+      "type": "string"
+    },
+    "masterAvailabilitySetID": {
+      "type": "string"
+    },
+    "masterVMSize": {
+      "type": "string"
+    },
+    "masterDataDiskSizeGB": {
+      "type": "int"
+    },
+    "masterAdminUsername": {
+      "type": "string"
+    },
+    "masterDisablePasswordAuth": {
+      "type": "bool",
+      "defaultValue": true
+    },
+    "masterAdminSSHKeyData": {
+      "type": "string"
+    },
+    "masterOSImagePublisher": {
+      "type": "string"
+    },
+    "masterOSImageOffer": {
+      "type": "string"
+    },
+    "masterOSImageSKU": {
+      "type": "string"
+    },
+    "masterOSImageVersion": {
+      "type": "string"
+    },
+    "masterCloudConfigData": {
+      "type": "string"
+    }
+  },
+  "variables": {
+    "masterAdminSSHKeyPath": "[concat('/home/',parameters('masterAdminUsername'),'/.ssh/authorized_keys')]",
+    "masterNICName": "[concat(parameters('clusterID'), '-MasterNIC-')]",
+    "masterVMName": "[concat(parameters('clusterID'), '-Master-')]",
+    "masterOSDiskName": "[concat(parameters('clusterID'), '-OSDisk-')]",
+    "masterDataDiskName": "[concat(parameters('clusterID'), '-DataDisk-')]"
+  },
+  "resources": [
+    {
+      "apiVersion": "[parameters('networkInterfacesAPIVersion')]",
+      "type": "Microsoft.Network/networkInterfaces",
+      "name": "[concat(variables('masterNICName'), copyIndex())]",
+      "location": "[resourceGroup().location]",
+      "copy": {
+        "name": "masterNICCopy",
+        "count": "[parameters('masterCount')]"
+      },
+      "properties": {
+        "ipConfigurations": [
+          {
+            "name": "MastersIpConfig",
+            "properties": {
+              "privateIPAllocationMethod": "Dynamic",
+              "subnet": {
+                "id": "[parameters('masterSubnetID')]"
+              }
+            }
+          }
+        ]
+      }
+    },
+    {
+      "apiVersion": "[parameters('virtualMachinesAPIVersion')]",
+      "type": "Microsoft.Compute/virtualMachines",
```

Comment:
Shouldn't we use some kind of autoscaling here?

Reply:
I had a talk with @rossf7 a couple of days ago and the conclusion was that for now, we would only run with 1 master.

Although, this PR allows the create of more than one master is desired (masterCount and masterIPAddresses parameters).

I propose we attack the autoscaling into a separate PR/issue. What do you say?

Reply:
Cool